internal/vault/secrets/ephemeral/gcp_oauth2_access_token.go

```diff
@@ -0,0 +1,321 @@
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/base"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/client"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/errutil"
+	"github.com/hashicorp/vault/api"
+)
+
+// Ensure the implementation satisfies the ephemeral.EphemeralResource interface
+var _ ephemeral.EphemeralResource = &GCPOAuth2AccessTokenEphemeralResource{}
+
+// NewGCPOAuth2AccessTokenEphemeralResource returns the implementation for this resource to be
+// imported by the Terraform Plugin Framework provider
+var NewGCPOAuth2AccessTokenEphemeralResource = func() ephemeral.EphemeralResource {
+	return &GCPOAuth2AccessTokenEphemeralResource{}
+}
+
+// GCPOAuth2AccessTokenEphemeralResource implements the methods that define this resource
+type GCPOAuth2AccessTokenEphemeralResource struct {
+	base.EphemeralResourceWithConfigure
+}
+
+// GCPOAuth2AccessTokenModel describes the Terraform resource data model to match the
+// resource schema.
+type GCPOAuth2AccessTokenModel struct {
+	// common fields to all ephemeral resources
+	base.BaseModelEphemeral
+
+	// fields specific to this resource
+	Backend             types.String `tfsdk:"backend"`
+	Roleset             types.String `tfsdk:"roleset"`
+	StaticAccount       types.String `tfsdk:"static_account"`
+	ImpersonatedAccount types.String `tfsdk:"impersonated_account"`
+
+	// computed fields
+	Token          types.String `tfsdk:"token"`
+	TokenTTL       types.Int64  `tfsdk:"token_ttl"`
+	LeaseID        types.String `tfsdk:"lease_id"`
+	LeaseDuration  types.Int64  `tfsdk:"lease_duration"`
+	LeaseStartTime types.String `tfsdk:"lease_start_time"`
+	LeaseRenewable types.Bool   `tfsdk:"lease_renewable"`
+}
+
+// Schema defines this resource's schema which is the data that is available in
+// the resource's configuration, plan, and state
+func (r *GCPOAuth2AccessTokenEphemeralResource) Schema(_ context.Context, _ ephemeral.SchemaRequest, resp *ephemeral.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"backend": schema.StringAttribute{
+				MarkdownDescription: "GCP Secret Backend to read credentials from.",
+				Required:            true,
+			},
+			"roleset": schema.StringAttribute{
+				MarkdownDescription: "GCP Secret Roleset to generate OAuth2 access token for. Mutually exclusive with `static_account` and `impersonated_account`.",
+				Optional:            true,
+			},
+			"static_account": schema.StringAttribute{
+				MarkdownDescription: "GCP Secret Static Account to generate OAuth2 access token for. Mutually exclusive with `roleset` and `impersonated_account`.",
+				Optional:            true,
+			},
+			"impersonated_account": schema.StringAttribute{
+				MarkdownDescription: "GCP Secret Impersonated Account to generate OAuth2 access token for. Mutually exclusive with `roleset` and `static_account`.",
+				Optional:            true,
+			},
+			"token": schema.StringAttribute{
```

Comment:
I believe we already have this field as a constant; would you mind updating accordingly. Thanks!